File: misc/productexceptself/bruteforce/ProductExceptSelf.py
class ProductExceptSelf:
    def productExceptSelf(self, nums):
        n = len(nums)
        result = [1] * n

        # For each element, compute the product of all other elements
        for i in range(n):
            product = 1
            for j in range(n):
                if i != j:
                    product *= nums[j]
            result[i] = product

        return result

# Example usage
solution = ProductExceptSelf()
nums = [1, 2, 3, 4]
print(solution.productExceptSelf(nums))  # Output: [24, 12, 8, 6]

File: misc/productexceptself/bruteforce/ProductExceptSelf.java
package productexceptself.bruteforce;

public class ProductExceptSelf {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;
        int[] result = new int[n];

        // For each element in nums, calculate the product of all other elements
        for (int i = 0; i < n; i++) {
            int product = 1;
            for (int j = 0; j < n; j++) {
                if (i != j) {
                    product *= nums[j];
                }
            }
            result[i] = product;
        }

        return result;
    }

    public static void main(String[] args) {
        ProductExceptSelf solution = new ProductExceptSelf();
        int[] nums = {1, 2, 3, 4};
        int[] result = solution.productExceptSelf(nums);

        // Print result
        for (int num : result) {
            System.out.print(num + " ");
        }
    }
}

